### PR TITLE
ASDF serialization for `MapAxes` class

### LINKED ISSUE #6713: ASDF: Implementation of phase 1 `MapAxes`
This is Phase 1 for Serialization Of Map and Model Classes into ASDF format as part of GSoC 2026 Project.

Implement `MapAxes` schema, converter for ASDF serialization and tests to verify read/write and schema validation.
By closing this issue it will be the end of Phase 1 ( all  `gammapy.maps.axes ` classes is serialized).

### PR DESCRIPTION
**Pull request by Basmala92**
**Description**

This pull request is part of Google Summer of Code project Serialization Of Map and Model Classes into ASDF (Advanced Scientific Data Format).

It closes #6713  which is serialization of `gammapy.maps.MapAxes` class which is a list of map axis objects into ASDF.

**Implementation details**

- The schema `mapaxes-1.0.0.yaml` to validate MapAxes fields which is a list of any map axis class ( MapAxis - TimeMapAxis - LabelMapAxis) that is already serialized and taged into the schema.
- `MapAxesConverter`: converts MapAxes object from/to ASDF.
- Added tests in `test_axes.py` to test the validate schema and round-trip tests.
- Registered `MapAxesConverter` in `extensions.py`.
- Updated `gammapy-1.0.0.yaml` with  MapAxes schema and tag.

### CODE REVIEW

File: gammapy/io/asdf/converters/maps/tests/test_axes.py [hunk 1]
```diff
@@ -287,3 +287,125 @@ def test_label_map_axis_read_examples(example):
     else:
         with pytest.raises(asdf.exceptions.ValidationError):
             asdf.open(buff)
+
+
+tested_map_axes = [
```
Comment (registerrier) [inline gammapy/io/asdf/converters/maps/tests/test_axes.py:292]: you might test the roundtrip with an empty [] MapAxes as well.

File: gammapy/io/asdf/converters/maps/axes.py [hunk 2]
```diff
@@ -83,3 +83,24 @@ def from_yaml_tree(self, node, tag, ctx):
             labels=node["labels"],
             name=node.get("name", ""),
         )
+
+
+class MapAxesConverter(Converter):
```
Comment (registerrier) [inline gammapy/io/asdf/converters/maps/axes.py:88]: MapAxes contains a second argument: n_spatial_axes
It is barely used except in the constructor of the `RegionGeom` where it is forced to 2. In general it is not serialized but recreated on the Geom init. 
So we probably don't need it in general. Now given the ASDF concept it might be bad not to export an actual attribute of the class.

File: gammapy/io/asdf/resources/schemas/maps/mapaxes-1.0.0.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    type: array
+    items:
+      anyOf:
+        - tag: "asdf://gammapy.org/gammapy/tags/maps/mapaxis-1.0.0"
+        - tag: "asdf://gammapy.org/gammapy/tags/maps/timemapaxis-1.0.0"
+        - tag: "asdf://gammapy.org/gammapy/tags/maps/labelmapaxis-1.0.0"
+    description: |
+      List of map axis objects.
+
+required : [axes]
```
Comment (registerrier) [inline gammapy/io/asdf/resources/schemas/maps/mapaxes-1.0.0.yaml:24]: what about n_spatial_axes?

**Review by registerrier — COMMENTED** [review]
Thanks @Basmala92 . That's good.

I have one suggestion and one concern:
- maybe test with an empty MapAxes, as this can be a common case
- there is a MapAxis.__init__ which you have not considered: n_spatial_axes. See inline comment. Is there a specific reason?

**Comment by Basmala92** [comment]
@registerrier Thank you for the review!

-  Ok will add the test for empty axes.

- Based on ASDF concept we don't need to include every attribute just the ones that is required to reconstruct the object, here `n_spatial_axes` is defaulted to None and as you said it is not commonly used and forced to a value in `Geom` classes, That is why i didn't include it.
Do you think it is better to add it as optional and check in the converter if it is not None?

File: gammapy/io/asdf/converters/maps/axes.py (hunk 2 above)
Comment (Astro-Kirsty) [reply gammapy/io/asdf/converters/maps/axes.py:88]: It was decided to not include the `n_spatial_axes` as it might have odd effects if you want to reuse the MapAxes for a WcsGeom

File: gammapy/io/asdf/converters/maps/axes.py [hunk 4]
```diff
@@ -83,3 +83,24 @@ def from_yaml_tree(self, node, tag, ctx):
...
+    """ASDF converter for the MapAxes class."""
```
Comment (Astro-Kirsty) [inline gammapy/io/asdf/converters/maps/axes.py]: Sorry I didn't pick up on this before for the other PRs but we should have the correct link here 
```suggestion
    """ASDF converter for the `~gammapy.maps.MapAxes` class."""
```

Comment (Astro-Kirsty) [reply gammapy/io/asdf/converters/maps/axes.py]: This could be implemented throughout

File: gammapy/io/asdf/resources/schemas/maps/mapaxes-1.0.0.yaml (hunk 3 above)
Comment (Astro-Kirsty) [reply gammapy/io/asdf/resources/schemas/maps/mapaxes-1.0.0.yaml:24]: See previous comment

**Review by Astro-Kirsty — COMMENTED** [review]
Thank you @Basmala92, just some small comments

**Comment by Basmala92** [comment]
> Thank you @Basmala92, just some small comments

Thanks Kirsty, Done!
